src/metabase/driver/sql_jdbc/actions.clj

```diff
@@ -15,32 +15,57 @@
                     :status-code status-code}
                    more-info))))
 
+(defn- check-one-row-effected [conn query raw-hsql] ;
+  (let [select-hsql     (-> raw-hsql (assoc :select [[:%count.* :row-count]]))
+        row-count       (:row_count (first (jdbc/query conn (hformat/format select-hsql))) 0)]
+    (when (not= 1 row-count)
+      (throw (ex-info (i18n/tru "Sorry, this would effect {0} rows, but you can only act on 1" row-count)
+                      {:query       query
+                       :sql         select-hsql
+                       :status-code 400})))))
+
 (defmethod actions/row-action! [:delete :sql-jdbc]
   [_action driver {database-id :database :as query}]
   (let [connection-spec (sql-jdbc.conn/db->pooled-connection-spec database-id)
         raw-hsql        (qp.store/with-store
                           (try
                             (qp/preprocess query) ; seeds qp store as a side effect so we can generate honeysql
-                            (catch Exception e
-                              (catch-throw e 404)))
-                          (try
                             (sql.qp/mbql->honeysql driver query)
                             (catch Exception e
-                              (catch-throw e 404))))
-        select-hsql     (-> raw-hsql (assoc :select [[:%count.* :row-count]]))
-        row-count       (:row_count (first (jdbc/query connection-spec (hformat/format select-hsql))) 0)]
-    (when (not= 1 row-count)
-      (throw (ex-info (i18n/tru "Sorry, this would delete {0} rows, but you can only delete 1 and only 1!!!" row-count)
-                      {:query       query
-                       :sql         select-hsql
-                       :status-code 400})))
-    (let [delete-hsql (-> raw-hsql (dissoc :select) (assoc :delete []))
+                              (catch-throw e 404))))]
+    (check-one-row-effected connection-spec query raw-hsql)
```

Comment:
Would it be better to call this `check-one-row-effected!` ?

Reply:
I'd say no bc it doesn't make any side-effect